pkg/services/syncer/syncer.go

```diff
@@ -106,44 +105,24 @@ func syncNamespace(config config.Config, namespace string) error {
 			continue
 		}
 
-		if _, ok := versions[vc.Namespace]; !ok {
-			versions[vc.Namespace] = []provv1.ManagedOSVersion{}
-		}
-
-		blockDel := false
 		for _, v := range vers {
-			vcpy := v.DeepCopy()
-			vcpy.ObjectMeta.Namespace = vc.Namespace
```

Comment:
we do this for several reasons:
- to overwrite whatever namespace is passed by the discoverer - ideally the discoverer should pass by clean objects, but we should be defensive enough to not allow the namespace to be overwritten from what we discover
- to stick 1:1 correlation. it would be weird that a ManagedOSVersionChannel from namespace A would be populating resources to B

Reply:
so there could be a wrong namespace in the version being created that doesnt match the current namespace?

Ok, in that case I think I will add a guard in here to check the namespace and create a new vcpy with the proper namespace if they dont match.